icu4c/source/i18n/rbnf.cpp

```diff
@@ -1700,9 +1700,9 @@ RuleBasedNumberFormat::setContext(UDisplayContext value, UErrorCode& status)
         if ( capitalizationBrkIter == NULL && (value==UDISPCTX_CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE ||
                 (value==UDISPCTX_CAPITALIZATION_FOR_UI_LIST_OR_MENU && capitalizationForUIListMenu) ||
                 (value==UDISPCTX_CAPITALIZATION_FOR_STANDALONE && capitalizationForStandAlone)) ) {
-            UErrorCode status = U_ZERO_ERROR;
-            capitalizationBrkIter = BreakIterator::createSentenceInstance(locale, status);
-            if (U_FAILURE(status)) {
+            UErrorCode status2 = U_ZERO_ERROR;
```

Comment:
Similar to the other cases, this could also potentially fail due to an OOM error. I think it might also be better to reuse the existing `status` parameter from the caller.